atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/WriteInfoTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static com.palantir.atlasdb.sweep.queue.WriteInfoPartitioner.SHARDS;
+
+import org.junit.Test;
+
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.sweep.Sweeper;
+
+public class WriteInfoTest {
+    private static final TableReference TABLE_REF = TableReference.createFromFullyQualifiedName("test.test");
+    private static final Cell CELL = Cell.create(new byte[]{1}, new byte[]{2});
+    private static final long ONE = 1L;
+    private static final long TWO = 2L;
+
+    @Test
+    public void tombstoneStatusIsIgnoredForEqualityAndSharding() {
```

Comment:
Why is this desirable?

Reply:
We need it to be the same for sharding. The equality is inherited from WriteReference, where the tombstone status is an auxiliary variable to simplify the logic since a tombstone and a normal write to the same cell are equivalent for the purposes of grouping writes and determining the one with the highest timestamp.